### PR TITLE
document how to setup IAM roles and service accounts

### PR DESCRIPTION
**Pull request by linki**
documents cloud provider setup in response to https://github.com/zalando-incubator/mate/issues/36

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,6 +65,143 @@ $ mate \
...
 
+### Permissions
+
+`Mate` needs permission to modify DNS records in your chosen cloud provider.
+On GCP this maps to using service accounts and scopes, on AWS to IAM roles and policies.
+
+#### AWS
+
+By default, `mate` runs under the same IAM role as the cluster node that the instance
+of `mate` currently runs on. If you want to resctrict the permissions `mate` gets you can use [`kube2iam`](https://github.com/jtblin/kube2iam) to set a different IAM role as context for each of your Pods in general.
```
Comment (Raffo) [inline README.md]: resctrict -> restrict

Comment (Raffo) [reply README.md]: add a `,` after "gets"

File: README.md [hunk 2]
```diff
@@ -65,6 +65,143 @@ $ mate \
...
+
+Follow the instructions in the `kube2iam` docs on how to deploy it in your cluster.
```
Comment (Raffo) [inline README.md]: Add link to `kube2iam` also here.

**Comment by ideahitme** [comment]
@linki not sure if kube2iam specific configuration is really necessary here. It floods the Readme file and probably just mentioning kube2iam as one of the possible implementation could be enough

**Comment by paultiplady** [comment]
For GCP, I think this is the minimum viable role: https://cloud.google.com/dns/access-control `roles/dns.admin`.

Note that this isn't configurable in the IAM console webapp, you need to use `gcloud` to assign it to a service account.

**Comment by linki** [comment]
@ideahitme I moved the description on how to setup correct IAM permissions to the examples and made a reference to it from the main README.

**Comment by ideahitme** [comment]
👍

**Pull request merged**